### PR TITLE
add button to copy qrCode text

### PR DESCRIPTION
**Pull request by GuillaumeZaki**
I’ve added a button that allows users to copy the text generated for the QR code. When the text is successfully copied, a message will appear to confirm the action. This is my first contribution

You can find the changes below:

Capture d’écran 2024-11-04 à 12 41 44
Capture d’écran 2024-11-04 à 12 42 30

### CODE REVIEW

**Comment by rugk** [comment]
Hi @GuillaumeZaki,
first of all, thanks for your first contribution to this project! :tada: :+1: 🏅
I hope you'll like this project and enjoy hacking on it… :smiley:

I like how you started contributing and it all seems to work. The question just stands, whether this feature is actually useful/wanted by the users of the extension.

Is there any issue that may be related where users asked for this or so? What is the actual use case behind this? Which problem does it solve?

Because from my experience, I see no reason to copy the _text_ there from that menu. If you select a text and open the QR Code generator from the context menu, you could just directly click "copy". If you use the URL as your example demonstrates you could also just right-click in the browser's address bar and click "copy" there.

We had a similar request in https://github.com/rugk/offline-qr-code/issues/285 for copying the SVG _image_ but that is something totally different. There I can see the usefulness, after all…

I am sorry I cannot give a more positive response, I hope the experience of contributing helped you anyway and you've learned something along the way…

File: src/popup/qrcode.css [hunk 1]
```diff
@@ -125,3 +125,34 @@ html, body {
     color: var(--white-100);
   }
 }
+
+/* Copy button  */
```
Comment (rugk) [inline src/popup/qrcode.css:129]: While this works, it does not look very nice. I would have suggested to style the elements according to (now obsolete but still used here) Firefox Photon style guide like https://web.archive.org/web/20220519062052/https://design.firefox.com/photon/components/message-bars.html

Also maybe a success message using https://github.com/TinyWebEx/MessageHandler (already included here) could have been used instead, which would integrate it more into the existing style.

**Review by rugk — CHANGES_REQUESTED** [review]
I've reviewed it anyway to give you tips for the next time, but I've decided not to merge it. I am sorry, but the idea of the UI/popup interface there is to be as slim as possible in order not to do the one thing it should do: show/generate a QR code. There is no "generate" button by purpose, e.g.

**Comment by GuillaumeZaki** [comment]
Hi Rugk,

Thank you for taking the time to review my pull request and for all the helpful feedback! I appreciate your detailed insights on the feature's utility and the suggestions you gave me for future contributions. This was actually my first contribution to an open-source project, so your advice is especially valuable as I work on improving my skills. 😁

Thanks again for your time and for welcoming new contributors like me. I’m looking forward to contributing more in the future!

**Pull request closed without merging**